File: app/src/main/java/com/mlsdev/mlsdevstore/presentaion/account/AccountViewModel.java
package com.mlsdev.mlsdevstore.presentaion.account;

import android.arch.lifecycle.Lifecycle;
import android.arch.lifecycle.OnLifecycleEvent;
import android.content.Context;
import android.databinding.ObservableField;

import com.mlsdev.mlsdevstore.data.local.LocalDataSource;
import com.mlsdev.mlsdevstore.data.model.order.GuestCheckoutSessionRequest;
import com.mlsdev.mlsdevstore.data.remote.BaseObserver;
import com.mlsdev.mlsdevstore.presentaion.viewmodel.BaseViewModel;

import javax.inject.Inject;

public class AccountViewModel extends BaseViewModel {
    private Context context;
    private LocalDataSource dataSource;

    // Personal info
    public final ObservableField<String> email = new ObservableField<>();
    public final ObservableField<String> firstName = new ObservableField<>();
    public final ObservableField<String> lastName = new ObservableField<>();

    // Shipping info
    public final ObservableField<String> phoneNumber = new ObservableField<>();
    public final ObservableField<String> address = new ObservableField<>();
    public final ObservableField<String> city = new ObservableField<>();
    public final ObservableField<String> state = new ObservableField<>();
    public final ObservableField<String> zip = new ObservableField<>();

    // Payment info
    public final ObservableField<String> cardNumber = new ObservableField<>();

    @Inject
    public AccountViewModel(Context context, LocalDataSource dataSource) {
        this.context = context;
        this.dataSource = dataSource;
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_START)
    void start() {
        dataSource.getGuestCheckoutSession()
                .subscribe(new BaseObserver<GuestCheckoutSessionRequest>(this) {
                    @Override
                    public void onSuccess(GuestCheckoutSessionRequest data) {
                        super.onSuccess(data);
                        email.set(data.getContactEmail() != null ? data.getContactEmail() : "");
                        firstName.set(data.getContactFirstName() != null ? data.getContactFirstName() : "");
                        lastName.set(data.getContactLastName() != null ? data.getContactLastName() : "");
                    }
                });
    }

    public void onEditPersonalInfoClick() {
        // TODO: 3/6/18 start personal info editing
    }

    public void onEditShippingInfoClick() {
        // TODO: 3/6/18 start shipping info editing
    }

    public void onEditPaymentInfoClick() {
        // TODO: 3/6/18 start payment info editing
    }
}

File: app/src/main/java/com/mlsdev/mlsdevstore/presentaion/account/AccountViewModel.kt
package com.mlsdev.mlsdevstore.presentaion.account

import android.content.Context
import androidx.databinding.ObservableField
import androidx.lifecycle.Lifecycle
import androidx.lifecycle.OnLifecycleEvent
import com.mlsdev.mlsdevstore.data.local.LocalDataSource
import com.mlsdev.mlsdevstore.presentaion.utils.Utils
import com.mlsdev.mlsdevstore.presentaion.viewmodel.BaseViewModel
import javax.inject.Inject

class AccountViewModel @Inject
constructor(
        context: Context,
        utils: Utils,
        private val localDataSource: LocalDataSource) : BaseViewModel() {

    init {
        this.context = context
        this.utils = utils
    }

    // Personal info
    val email = ObservableField<String>()
    val firstName = ObservableField<String>()
    val lastName = ObservableField<String>()

    // Shipping info
    val phoneNumber = ObservableField<String>()
    val address = ObservableField<String>()
    val city = ObservableField<String>()
    val state = ObservableField<String>()
    val zip = ObservableField<String>()

    @OnLifecycleEvent(Lifecycle.Event.ON_START)
    internal fun start() {
        checkNetworkConnection(utils!!) {
            compositeDisposable.add(localDataSource.getGuestCheckoutSession().subscribe(
                    { guestCheckoutSessionRequest ->
                        email.set(guestCheckoutSessionRequest.contactEmail)
                        firstName.set(guestCheckoutSessionRequest.contactFirstName)
                        lastName.set(guestCheckoutSessionRequest.contactLastName)
                        phoneNumber.set(guestCheckoutSessionRequest.shippingAddress.phoneNumber)
                        address.set(guestCheckoutSessionRequest.shippingAddress.address)
                        city.set(guestCheckoutSessionRequest.shippingAddress.city)
                        state.set(guestCheckoutSessionRequest.shippingAddress.state)
                        zip.set(guestCheckoutSessionRequest.shippingAddress.postalCode)
                    },
                    { handleError(it) }))
        }
    }
}
